packages/dht/README.md

```diff
@@ -1,17 +1,99 @@
-# @streamr/dht (WIP)
+# @streamr/dht
 
-An implementation of the Kademlia-based Streamr DHT used for the trackerless network.
+A connectionful Kademlia based P2P distributed hash table (DHT). Implements an in-memory key-value data store where multiple creators of of data can list values behind a single key. The library is entirely browser compatible.
 
-## Generating Protobuf code
+All communication between peers utilizes protobuf, transported with the help of the custom `@streamr/proto-rpc` library.
 
-Auto-generated protobuf/gRPC code is compiled during the build process.
-```bash
-  npm run build
+Connections in the DHT are established using WebSocket or WebRTC connections. The nodes will decide internally based on peers' connectivity information which connection types to use.
+
+The DHT also provides an interface for sending messages from peer to peer with the `DhtNode#send` function. The sent message is routed over the network to the target node's id.
+
+## Running a node
+
+Running a bare DhtNode in the Streamr Network is heavily discouraged as it requires advanced configuration. However, if you do wish to run a node here are some recommended configurations.
+
+(the following values are examples for working production or test values check the streamr-client's default network configuration)
+
+```js
+const DhtNode = new DhtNode({
+  // Add a list of known entry points to the network
+  entryPoints: [{
+    kademliaId: new Uint8Array([1, 2, 3])
+  }], 
+  // a list of STUN and TURN servers. Critically important when opening webrtc conenctions behind NATs.
+  iceServers: [{
+    url: "stun.l.google.com",
+    port: 19302
+  }],
+  // A range of ports that the node will attempt to start a WebSocket server in. If you wish to use an exact port give equal values to min and max.
+  websocketPortRange: {
+    min: 30000,
+    max: 30500
+  },
+  ...
+})
+```
+
+## WebSocket Server TLS configuration
+
+Setting up TLS is important 
+
+There two ways to configure TLS for a DhtNode's WS server.
+
+### Auto-certification
+
+When starting a node configure it to automatically fetch TLS certificates for a randomly generated domain name as such:
+
+```js
+const DhtNode = new DhtNode({
+  ...
+  websocketServerEnableTls: true,
```

Comment:
Not related to this PR: maybe `autoCertified* `config options could be moved into the `tlsCertificate` block in the future?

Reply:
yeah at some point configuration related to specific features could be given their specific objects